pkg/skaffold/build/docker.go

```diff
@@ -31,12 +34,36 @@ import (
 func (l *LocalBuilder) buildDocker(ctx context.Context, out io.Writer, workspace string, a *v1alpha2.DockerArtifact) (string, error) {
 	initialTag := util.RandomID()
 
-	err := docker.RunBuild(ctx, out, l.api, workspace, types.ImageBuildOptions{
-		Tags:       []string{initialTag},
-		Dockerfile: a.DockerfilePath,
-		BuildArgs:  a.BuildArgs,
-		CacheFrom:  a.CacheFrom,
-	})
+	var err error
+	if l.cfg.UseDockerCLI || l.cfg.UseBuildkit {
+		absDockerfilePath := a.DockerfilePath
+		if !filepath.IsAbs(a.DockerfilePath) {
+			absDockerfilePath = filepath.Join(workspace, a.DockerfilePath)
```

Comment:
I think you also need to be sure you don't append the workspace twice, since the workspace itself can also be a relative path. I did this [here](https://github.com/GoogleContainerTools/skaffold/blob/master/cmd/skaffold/app/cmd/docker/deps.go#L82), you could pull this out into a util package and reuse it here.

Reply:
Good point. I'll do that